pkg/services/pluginsintegration/clientmiddleware/clear_auth_headers_middleware_test.go

```diff
@@ -73,34 +109,76 @@ func TestClearAuthHeadersMiddleware(t *testing.T) {
 					PluginContext: pluginCtx,
 					Headers:       map[string]string{otherHeader: "test"},
 				})
 				require.NoError(t, err)
 				require.NotNil(t, cdt.QueryDataReq)
 				require.Len(t, cdt.QueryDataReq.Headers, 1)
+				require.Equal(t, "test", cdt.QueryDataReq.Headers[otherHeader])
+				require.Empty(t, cdt.QueryDataReq.GetHTTPHeaders())
 			})
 
-			t.Run("Should not attach delete headers middleware when calling CallResource", func(t *testing.T) {
+			t.Run("No auth heads to clear when calling CallResource", func(t *testing.T) {
 				err = cdt.MiddlewareHandler.CallResource(req.Context(), &backend.CallResourceRequest{
 					PluginContext: pluginCtx,
 					Headers:       map[string][]string{otherHeader: {"test"}},
 				}, nopCallResourceSender)
 				require.NoError(t, err)
 				require.NotNil(t, cdt.CallResourceReq)
 				require.Len(t, cdt.CallResourceReq.Headers, 1)
+				require.Equal(t, []string{"test"}, cdt.CallResourceReq.Headers[otherHeader])
+				require.Equal(t, http.Header{http.CanonicalHeaderKey(otherHeader): {"test"}}, cdt.CallResourceReq.GetHTTPHeaders())
 			})
 
-			t.Run("Should not attach delete headers middleware when calling CheckHealth", func(t *testing.T) {
+			t.Run("No auth heads to clear when calling CheckHealth", func(t *testing.T) {
 				_, err = cdt.MiddlewareHandler.CheckHealth(req.Context(), &backend.CheckHealthRequest{
 					PluginContext: pluginCtx,
 					Headers:       map[string]string{otherHeader: "test"},
 				})
 				require.NoError(t, err)
 				require.NotNil(t, cdt.CheckHealthReq)
 				require.Len(t, cdt.CheckHealthReq.Headers, 1)
+				require.Equal(t, "test", cdt.CheckHealthReq.Headers[otherHeader])
+				require.Empty(t, cdt.CheckHealthReq.GetHTTPHeaders())
+			})
+
+			t.Run("No auth heads to clear when calling SubscribeStream", func(t *testing.T) {
+				_, err = cdt.MiddlewareHandler.SubscribeStream(req.Context(), &backend.SubscribeStreamRequest{
+					PluginContext: pluginCtx,
+					Headers:       map[string]string{otherHeader: "test"},
+				})
+				require.NoError(t, err)
+				require.NotNil(t, cdt.SubscribeStreamReq)
+				require.Len(t, cdt.SubscribeStreamReq.Headers, 1)
+				require.Equal(t, "test", cdt.SubscribeStreamReq.Headers[otherHeader])
+				require.Empty(t, cdt.SubscribeStreamReq.GetHTTPHeaders())
+			})
+
+			t.Run("No auth heads to clear when calling PublishStream", func(t *testing.T) {
+				_, err = cdt.MiddlewareHandler.PublishStream(req.Context(), &backend.PublishStreamRequest{
+					PluginContext: pluginCtx,
+					Headers:       map[string]string{otherHeader: "test"},
+				})
+				require.NoError(t, err)
+				require.NotNil(t, cdt.PublishStreamReq)
+				require.Len(t, cdt.PublishStreamReq.Headers, 1)
+				require.Equal(t, "test", cdt.PublishStreamReq.Headers[otherHeader])
+				require.Empty(t, cdt.PublishStreamReq.GetHTTPHeaders())
+			})
+
+			t.Run("No auth heads to clear when calling RunStream", func(t *testing.T) {
```

Comment:
nitpick: Maybe write headers instead of head
```suggestion
			t.Run("No auth headers to clear when calling RunStream", func(t *testing.T) {
```